src/cascadia/TerminalSettingsModel/IInheritable.h

```diff
@@ -206,3 +211,57 @@ private:                                                                    \
         /*no value was found*/                                              \
         return std::nullopt;                                                \
     }
+
+// This macro is similar to the ones above, but is reserved for settings
+// that are arrays (like color table)
+#define GETSET_ARRAY_SETTING(type, size, name)                              \
+public:                                                                     \
+    /* Returns true if the user explicitly set the value, false otherwise*/ \
+    bool Has##name() const                                                  \
+    {                                                                       \
+        return _##name.has_value();                                         \
+    }                                                                       \
+                                                                            \
+    /* Returns the resolved value for this setting */                       \
+    /* fallback: user set value --> inherited value --> empty array */      \
+    std::array<type, size> name() const                                     \
+    {                                                                       \
+        const auto val{ _get##name##Impl() };                               \
+        return val ? *val : std::array<type, size>{};                       \
+    }                                                                       \
+                                                                            \
+    /* Overwrite the user set value */                                      \
+    void name(const std::array<type, size>& value)                          \
+    {                                                                       \
+        _##name = value;                                                    \
+    }                                                                       \
+                                                                            \
+    /* Clear the user set value */                                          \
+    void Clear##name()                                                      \
+    {                                                                       \
+        _##name = std::nullopt;                                             \
+    }                                                                       \
+                                                                            \
+private:                                                                    \
+    std::optional<std::array<type, size>> _##name{ std::nullopt };          \
+    std::optional<std::array<type, size>> _get##name##Impl() const          \

```

Comment:
I wanna make sure Dustin takes a look at the template magic going on in this macro. An `optional<array<T>>` seems to make sense to me, but there might be weird template quirks I'm missing.